src/plugin/protocol-smb/src/java/org/apache/nutch/protocol/smb/Smb.java

```diff
@@ -0,0 +1,292 @@
+        LOG.warn("hostname={}", hostname);
+        LOG.warn("shareAndPath={}", shareAndPath);
+        components = shareAndPath.split("/", 2);
+        String shareName = components[0];
+        String path = components.length>1 ? "/" + components[1]: "/";
+        LOG.warn("share={}", shareName);
+        LOG.warn("path={}", path);
+
+        // todo: we construct and destruct the connection for each and every URL. Can connection pools improve?
+        SMBClient client = new SMBClient();
+        try(Connection connection = client.connect(hostname)) {
+
+          AuthenticationContext ac = new AuthenticationContext(user, password.toCharArray(), domain);
+          Session session = connection.authenticate(ac);
+
+          // Connect to Share
+          try (DiskShare share = (DiskShare) session.connectShare(shareName)) {
+
+            // now get the content
+            if (share.folderExists(path)) {
+              String c = getDirectoryContent(share, shareName, path);
+
+              String base = urlstr.toString();
+              if (base.endsWith("/")) {
+                base = base + ".";
+              }
+              if (!base.endsWith("/.")) {
+                base = base + "/.";
+              }
+
+              LOG.warn("base={}", base);
+              LOG.warn("directory={}", c);
+
+              return new ProtocolOutput(
+                new Content(base, base, c.getBytes(), "text/html", new Metadata(), getConf()), 
+                ProtocolStatus.STATUS_SUCCESS
+              );
+            } else if (share.fileExists(path)) {
+              // todo: how can we store this, and maybe more metadata?
+              Metadata metadata = new Metadata();
+              metadata.set(Metadata.CONTENT_TYPE, "application/octet-stream");
+
+              FileAllInformation fileInfo = share.getFileInformation(path);
+              File file = share.openFile(path, EnumSet.of(AccessMask.GENERIC_READ), null, SMB2ShareAccess.ALL, SMB2CreateDisposition.FILE_OPEN, null);
+
+              InputStream fileIn = file.getInputStream();
+              byte[] bytes = null;
+              long fileSize = fileInfo.getStandardInformation().getEndOfFile();
+              long fetchSize = fileSize;
+              metadata.add("fileSize", String.valueOf(fileSize));
+
+              // todo: we run into issues if the file is bigger than 2 GB. I made the limit configurable
+              // but e.g. zip can no longer be evaluated if too big.
+              if (fetchSize > contentLimit) {
+                LOG.warn("trunkating {}", urlstr);
+                fetchSize = contentLimit;
+
+                // todo: this metadata seems to be not available for the indexer. However it might be useful to know the content
+                // discovery is incomplete
+                metadata.add("truncated", String.valueOf(fetchSize));
+              }
+
+              bytes = IOUtils.toByteArray(fileIn, fetchSize); // read inputstream into byte array

```

Comment:
As I mentioned this one line is the only usage of `commons-io`, I think we can just use this from the `ivy/ivy.xml`.